README.md

```diff
@@ -447,6 +447,98 @@ or from the tenant resource (or from the application, in the case of account sto
   $accountStoreMapping->accountStore = $directory; // this could also be a group
   $application->createAccountStoreMapping($accountStoreMapping);
   ```
+  
+### Custom Data
+You may find it useful to store your own custom data on resources.  With the Stormpath SDK, we make it easy to do so. 
+All resources support a Custom Data collection for your own custom use.  The CustomData resource is a schema-less JSON 
+object (aka ‘map’, ‘associative array’, or ‘dictionary’) that allows you to specify whatever name/value pairs you wish.
+
+The CustomData resource is always connected to a resource and you can always reach it by calling the 
+getCustomData() method on the resource instance:
+
+```php
+  $application = \Stormpath\Resource\Application::get($applicationHref);
+
+  $applicationCustomData = $application->getCustomData();
+
+  //Or
+
+  $application = $client->
+                 dataStore->
+                 getResource($applicationHref, \Stormpath\Stormpath::APPLICATION);
+                 
+  $applicationCustomData = $application->getCustomData();
+
+  ```
+  
+In addition to your custom name/value pairs, a CustomData resource will always contain 3 reserved read-only fields:
+* href: The fully qualified location of the custom data resource
+* createdAt: the UTC timestamp with millisecond precision of when the resource was created in Stormpath as an ISO 8601 formatted string, for example 2017-04-01T14:35:16.235Z
+* modifiedAt: the UTC timestamp with millisecond precision of when the resource was last updated in Stormpath as an ISO 8601 formatted string.
+
+You can store an unlimited number of additional name/value pairs in the CustomData resource, with the following restrictions:
+
+* The total storage size of a single CustomData resource cannot exceed 10 MB (megabytes). The href, createdAt and modifiedAt field names and values do not count against your resource size quota.
+* Field names must:
+    * be 1 or more characters long, but less than or equal to 255 characters long (1 <= N <= 255).
+    * contain only alphanumeric characters 0-9A-Za-z, underscores _ or dashes - but cannot start with a dash -.
+    * may not equal any of the following reserved names: href, createdAt, modifiedAt, meta, spMeta, spmeta, ionmeta, or ionMeta.
+    
+#### Create Custom Data
+An example of creating an application with custom data
+```php 
+  $account = $client->dataStore->instantiate(\Stormpath\Stormpath::ACCOUNT);
+  $account->email = 'john.smith@example.com';
+  $account->givenName = 'John';
+  $account->password ='4P@$$w0rd!';
+  $account->surname = 'Smith';
+  $account->username = 'johnsmith';
+  
+  $customData = $account->getCustomData();
+  $customData->put("rank", "Captain");
+  $customData->put("birthDate", "2305-07-13");
+  $customData->put("favoriteDrink", "favoriteDrink");
```

Comment:
This is fine, but I think we could also have the calls be:

```php
$customData->favoriteDrink = "favoriteDrink";
```
This would imply a small change in the ```Magic.php``` file ("_set" method) that would be inherited by the new ```CustomData.php``` resource  class. This way we keep it consistent with other resources.